token/program-2022/src/extension/mod.rs

```diff
@@ -670,6 +670,8 @@ pub enum ExtensionType {
     ConfidentialTransferFeeAmount,
     /// Mint contains a pointer to another account (or the same account) that holds metadata
     MetadataPointer,
+    /// Mint contains token metadata
+    Metadata,
```

Comment:
Seems like slipping this in this PR at least justifies a separate commit. Maybe unintended?

Reply:
I wanted to have an error case to test, but I guess I could do a different one through a `#[cfg(test)]`

Reply:
Or you could just break into different commits ;)

Reply:
The `#[cfg(test)]` unsized variant seemed clearer, and it'll be much better for later